### ISSUE TITLE
Lint loop labels

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#72]
JavaScript loops can have labels, and `break` and `continue` statements can refer to those labels:
```javascript
outer: for (let y of ys) {
  for (let x of xs) {
    if (x == y) {
      break outer;
    }
  }
}
```

quick-lint-js currently doesn't ensure labels in `break` and `continue` statements refer to breakable or continuable loops. Teach the linter to detect problems related to loop labels, such as the following:

- [ ] `break`/`continue`: label doesn't exist at all in the current function
- [ ] `break`/`continue`: Label is attached to an unrelated loop
- [ ] label: Label is not attached to a loop statement (maybe not? doesn't seem like a serious problem to complain about by default.)

The parser doesn't tell the linter a lot of information needed to do this analysis (such as the labels themselves, or loop structures), so this is a decently involved task.

**Comment by erlliam** [comment]
I'd like to work on this issue.

**Comment by strager** [comment]
@erlliam I think you can assign the issue to yourself, since you're a member of quick-lint.

**Comment by strager** [comment]
@erlliam Can you share the research you've done on this task?

**Comment by erlliam** [comment]
My various thoughts and findings:

The most interesting thing:
`a: { break a; }` is valid.
`b: { continue b; }` throws an error in most JS runtimes.

`// Labelled statement.` is a comment that can be found in the code base. It is a good pointer to where labelled statement checks happen. Not sure if there's a colon check for yield/await.

To complete this issue you'd probably have to:
* Create undefined label error
* Create duplicate label error
* Create error for the code example above?
* Add a label set to the scope in the linter
* Add label assigment support to the linter
* Add label usage support to the linter
* Add something like "parse_and_visit_labelled_statement" to the parser
* Add label checking logic when parsing break/continue tokens

I'd say the solution would probably resemble the linter's variable code. If the issue is still open in a couple of weeks from now, I will return to it.

**Comment by strager** [comment]
I realized something: this task actually has two major parts:

1. Implement `break;` and `continue;`
2. Implement `break label;` and `continue label;`

Part 2 is pretty much a superset of part 1, so let's focus on implementing part 1 first. I filed a separate task for this: #151
